pyspeckit/spectrum/classes.py

```diff
@@ -291,17 +288,22 @@ def parse_text_header(self,Table):
         .txt or other atpy table type objects
         """
         self.Table = Table
+
         xtype = Table.data.dtype.names[Table.xaxcol]
         if xtype in units.xtype_dict.values():
             self.xarr.xtype = xtype
-            self.xarr.unit = Table.columns[xtype].unit
+            unit = Table.columns[xtype].unit
+            if unit == 'angstroms' : unit = 'angstrom'
+            self.xarr._unit = u.Unit(unit)
```

Comment:
should use `set_unit` here, and `set_unit` should handle the angstrom/angstroms issue